backend/countries/countries.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Dict, Generic, List, Optional, TypeVar, Union
+
+from countries.constants import CONTINENTS, COUNTRIES
+
+T = TypeVar("T", bound="Base")
+
+
+class Base(Generic[T]):
```

Comment:
I don't think I understand this code, can you explain what it is?

Reply:
I want the list of countries/continents to be "smarter" than a bare list of dictionaries. Since both Countries and Continents use the same logic I wrote a base class and use Generics for the typing...

Reply:
I want to mimic the behaviour of the PyCountries library and have the same api

Reply:
I am not a big fan of this code, we could just implement the API/interface that we need or a simpler one since our code is not a library, but up to you